### ISSUE TITLE
Deployed this zk-tree-vote project on testnet but encountered error

### ISSUE THREAD

**Issue by emmanbol** [TheBojda/zktree#4]
Hello, thanks for your earlier reply to the issue I raised.

The project worked perfectly on my local computer as expected.

Then, I decided to move it to testnet.

The voter validation aspect worked perfectly, but when to vote, it shows an error: "exceed maximum block range: 5000". I did some search, and I discover the error is due to "event" that is needed to be retrieved for computation. What can be done.

I think this code portion is the issue, there should be a way to mitigate. 
`export async function calculateMerkleRootAndPathFromEvents(mimc: any, address: any, provider: any, levels: number, element: any) {
    const abi = [
        "event Commit(bytes32 indexed commitment,uint32 leafIndex,uint256 timestamp)"
    ];
    const contract = new Contract(address, abi, provider)
    const events = await contract.queryFilter(contract.filters.Commit())
    let commitments = []
    for (let event of events) {
        commitments.push(BigNumber.from(event.args.commitment))
    }
    return calculateMerkleRootAndPath(mimc, levels, commitments, element)
}`

**Comment by TheBojda** [comment]
Can you send me the full stack trace? Which method throws this error?

**Comment by emmanbol** [comment]
I noticed the error when I click on "send to blockchain" on the Vote Page. After some Seconds, the console view on browser shows the error, nothing happens on the page itself. My metamask extension on browser is triggers the error. 

Is it possible to abandon event and just use the commitment and nullifer logic alone to process whatever. Seem event is big overhead and causes huge gas fees. Just thinking aloud. 

Note: I used bsc testnet, it may not be issue with ETH testnet or mainnet, but it is a issue.

**Comment by TheBojda** [comment]
I think, it is a restriction of the provider. Check this answer on STO: https://ethereum.stackexchange.com/questions/107590/contract-queryfilterfilter-giving-me-errors-in-ethers-js

**Comment by emmanbol** [comment]
Alright. Thanks. Appreciate.

**Comment by emmanbol** [comment]
Hello, see what I was thinking. 

I will Start another code, where what event is needed for will be stored on the contract. Then offchain, I can get this data, compute what is needed, then complete onchain. This way, I will bypass event and some of it's limitations. 

As it is, it seem to me, for each vote transaction, it gathers all the events from block 0 to latest block. That is a very big data. I may be wrong though. Educate me.

How can I use your present code, to do the above stated, because it seem your code depends solely on event. Even when I comment out the section, it still showed the event error. 

We can talk via telegram for easy communication. 

https://t.me/Defi_Solutions

**Comment by TheBojda** [comment]
Hi! 

The events are used to build the Merkle tree locally to calculate the Merkle proof. The code is in the calculateMerkleRootAndPathFromEvents method. It queries only the Commit events, and I think providers like Infura index these events, so I don't think that in this case, the provider reads all of the events from block 0. Btw, TC does the same thing with Deposit events (https://github.com/tornadocash/tornado-core/blob/master/src/cli.js). So, I think it should work in a production environment (by using Infura or other providers).

**Comment by emmanbol** [comment]
If I use infura for example, does it mean I will have to pay for the service if the free tier for example finishes. I have been doing blockchain apps without necessarily using custom providers. 

Also noticed, it is slow to pop the metamask app because of heavy computation. 

I changed to polygon test on metamask using alchemy api free tier, and it worked. 

But it delays before pop up, I think due to heavy computation.

**Comment by TheBojda** [comment]
Yep, building the tree and calculating the zkproof can be CPU intensive.

**Comment by emmanbol** [comment]
I agree with you 100%.

Can you think of a cheaper solution? I have fully understood the project, but the second part of securely validating and completing the interaction is the issue for me. I am thinking of a cheaper, cooler solution. Cos this is damn complex but it works though. 

It must have taken you some energy to come up with this cool tutorial and library. 

Haha.
